extension/extensionlimiter/README.md

```diff
@@ -0,0 +1,368 @@
+falling-back to consumerlimiter for limits that have not been enforced
+automatically.
+
+But, there are variations, like
+
+```yaml
+extensions:
+  ratelimiter/1:      # weights go here?
+  ratelimiter/3:      # weights go here?
+  admissionlimiter/2: # weights go here?
+  admissionlimiter/4: # weights go here?
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        middleware:
+        - ratelimiter/1
+        - admissionlimiter/2
+    limiters:
+    - ratelimiter/3    # multiple limiters here?
+    - admissionlimiter/4 # multiple limiters here?
+```
+
+The configuration leaves all weight configuration to the limiter
+components. Limiters can implement all weights, but they cannot be
+both a rate limiter and a resource limiter.
+
+Should we add indirection? Relax the requirement that a limiter cannot
+be both a rate and a resource limiter? In this example, indirection is
+added in the form of a "limitermux" extension. This can be an
+implementation detail.
+
+```yaml
+extensions:
+  ...
+  limitermux/grpc:
+    request_count:
+    - ratelimiter/1
+    network_bytes:
+    - ratelimiter/2
+    request_bytes:
+    - ratelimiter/3
+    - admissionlimiter/1
+    request_items:
+    - ratelimiter/4
+
+  limitermux/otlp:
+    ...
+
+  ...
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        middleware:
+        - limitermux/grpc
+    limiters:
+    - limitermux/otlp
+```
+
+Another route is to back up from the current PR, which makes
+configmiddleware.Config into a non-struct `component.ID` type,
+eliminating the struct-value with single `id:` field. Instead,
+we could add limiter-weight bindings to the middleware config
+and/or limiter config:
+
+```yaml
+extensions:
+  ...
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        middleware:
+        - network_bytes: ratelimiter/1
+          request_count: ratelimiter/2
+	  request_bytes: admissionlimiter/1
+      http:
+        middleware:
+        - network_bytes: ratelimiter/1
+          request_count: ratelimiter/2
+	- decompression: zstd,snappy,gzip
+        - request_bytes: admissionlimiter/1
+	- instrumentation: opentelemetry/1
+    limiters:
+    - request_items: ratelimiter/3
+```
+
+##### Instrumentation
+
+It is possible to extract Counter (RateLimiter) and UpDownCounter
+(ResourceLimiter) instrumentation to convey the rates and totals
+associated with each limiter.
+
+This should investigated along with investigation into the topics
+raised above. Users will be well served if we are able to confidently
+extract network-bytes, request-bytes, request-items, and request-count
+information without double-counting and provide instrumentation
+covering these variables.
+
+##### Are built-in rate and resource limiters needed?
+
+The provided helper implementations are based in
+`golang.org/x/time/rate` and
+`collector-contrib/internal/otelarrow/admission2`.  We could instead
+create two extension implementations for these. The code is a hundred
+lines or so each.
```

Comment:
IMO it's good to keep a clear line of separation between the extension interface and implementations.